auth/maintenance_socket_role_manager.cc

```diff
@@ -118,4 +119,13 @@ future<std::vector<cql3::description>> maintenance_socket_role_manager::describe
     return operation_not_supported_exception<std::vector<cql3::description>>("DESCRIBE SCHEMA WITH INTERNALS");
 }
 
+future<> maintenance_socket_role_manager::delete_maintenance_socket() {
+    std::error_code ec;
+    std::filesystem::remove("maintenance_socket", ec);
```

Comment:
This is a thread blocking API, which we're avoiding in the Seastar single-thread-per-core architecture.

Please use a nonblocking API from Seastar for removing files.